angular-register.ts

```diff
@@ -0,0 +1,30 @@
+declare interface NoopDecorator {
+  <T>(type: T): T;
+  (): (...args: any[]) => void;
+}
+
+declare global {
+  /**
+   * Indicates that this function or variable is being overridden
+   * from the implemented or extended parent class.
+   *
+   * @see [TSLint Override](https://github.com/hmil/tslint-override)
+   */
+  // tslint:disable-next-line: variable-name
+  const Override: NoopDecorator;
+
+  interface Window {
+    Override: NoopDecorator;
+  }
+
+  namespace NodeJS {
+      interface Global {
+          Override: NoopDecorator;
+      }
+  }
+}
+
+export const ctx = global;
```

Comment:
Granted I missed the browser-only case. But [globalThis](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) isn't standard yet and isn't well supported.

Might as well go for something like `typeof global === 'undefined' ? window : global`?
This would definitely work in the runtime, but at compile time, tsc might complain about `global` if `@types/node` is not installed, or complain about `window` if the DOM library is not loaded...

Or maybe I'm completely wrong and TypeScript provides the polyfill for globalThis?

Reply:
[TypeScript doesn't provide a pollyfill](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#type-checking-for-globalthis) for globalThis, however in my test on Edge Chromium, Opera, Chrome and Firefox, (on macOS), all return the `window` object from `globalThis` despite what [Can I Use](https://caniuse.com/#search=globalThis) and [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis) says. 

IE and the Original Edge are the only browsers I could think of that wouldn't support `globalThis` but wouldn't it be best to just use a pollyfill for the the time being to support those browsers until MS drops support for them?

Either that or just not support those browsers?